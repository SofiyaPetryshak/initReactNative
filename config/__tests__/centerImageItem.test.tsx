import React from 'react';
import renderer from 'react-test-renderer';
import CenterImageItem from '../../src/components/centerImageItem'

const mockItem = {
  id: 0,
  title: "ASOS DESIGN bag with detachable strap",
  brand: "VS",
  price: 100,
  description: "ASOS DESIGN rose gold zipped compartment tote bag with detachable strap",
  images: [
    "https://images.unsplash.com/photo-1596149615678-8488f200b301?ixid=MXwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2689&q=80",
    "https://images.unsplash.com/photo-1546188994-07c34f6e5e1b?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&auto=format&fit=crop&w=1951&q=80",
    "https://images.unsplash.com/photo-1547949003-9792a18a2601?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1950&q=80",
    "https://images.unsplash.com/photo-1527385352018-3c26dd6c3916?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1950&q=80"
  ],
  liked: false,
  color: "black",
  quantity: 0
}

 test('renders correctly', () => {
   const tree = renderer.create(<CenterImageItem item={mockItem} onPress={()=>{console.log('mock item')}} />).toJSON()
   expect(tree).toMatchSnapshot()
 });

// describe('is working', ()=> {
//   it('should work', ()=>{
//     expect(true).toBeTruthy()
//   })
// })

