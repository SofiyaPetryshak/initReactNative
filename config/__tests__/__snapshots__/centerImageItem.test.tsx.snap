// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "height": 290,
        "margin": 5,
        "overflow": "hidden",
        "width": 360,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "https://images.unsplash.com/photo-1596149615678-8488f200b301?ixid=MXwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2689&q=80",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "paddingVertical": 7,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        $
        100
      </Text>
      <Text />
    </View>
    <Text
      numberOfLines={2}
      style={
        Object {
          "paddingHorizontal": 10,
          "textAlign": "center",
        }
      }
    >
      ASOS DESIGN bag with detachable strap
    </Text>
  </View>
</View>
`;
